pkg/sentry/platform/kvm/machine_amd64.go

```diff
@@ -367,3 +368,41 @@ func rdonlyRegionsForSetMem() (phyRegions []physicalRegion) {
 func availableRegionsForSetMem() (phyRegions []physicalRegion) {
 	return physicalRegions
 }
+
+var execRegions []region
```

Comment:
Note that I think this should be moved into the mapUpperHalf function below. But if you're opposed to that, you should at least just do:

var execRegions = applyVirtualRegions(func(vr virtualRegion) {
   ...
})

.. And skip the extra init?

Reply:
mapUpperHalf() is called at every time when a pagetable is building, its performance is crucial.
And /proc/self/maps is changing when running, it will cause problems and maps too much unneeded region.
In near future(when re-implement #509), mapUpperHalf() will be only called once for a machine, we can optimize the extra init out that time. #509 is quite different topic, I don't want to implement it in this patch.

Reply:
Ah, I was actually just suggesting a code pattern that I think is a bit more robust. Instead of

var execRegions []region

func init() {
  ...
}

You can do:

var execRegions = func() (regions []region) {
   applyVirtualRegions(....)
}()

That way the init() won't be moved to modified and it's obviously bound to the execRegions global. This is just a minor style suggestion, it doesn't really impact the code. Up to you.